src/uct/ib/rc/verbs/rc_verbs_ep.c

```diff
@@ -646,6 +653,27 @@ ucs_status_t uct_rc_verbs_ep_flush(uct_ep_h tl_ep)
     return UCS_INPROGRESS;
 }
 
+ucs_status_t uct_rc_verbs_ep_fc_ctrl(uct_rc_ep_t *rc_ep)
+{
+    uct_rc_verbs_iface_t *iface = ucs_derived_of(rc_ep->super.super.iface,
+                                                 uct_rc_verbs_iface_t);
+    uct_rc_verbs_ep_t *ep = ucs_derived_of(rc_ep, uct_rc_verbs_ep_t);
+    uct_rc_hdr_t hdr;
+
+    ucs_assert(sizeof(hdr) <= iface->config.max_inline);
+    UCT_RC_CHECK_RES(&iface->super, &ep->super);
+
+    hdr.am_id                   = UCT_RC_EP_FC_PURE_GRANT;
+
+    iface->inl_sge[0].addr      = (uintptr_t)&hdr;
+    iface->inl_sge[0].length    = sizeof(hdr);
+    iface->inl_sge[1].addr      = (uintptr_t)NULL;
```

Comment:
let's pass a local work-request variable here and not reuse iface->inl_am_wr, because we have 1 sge and better to avoid passing (NULL/length=0) as 2nd sge